internal/qmp-gen/parse_test.go

```diff
@@ -79,3 +174,33 @@ Got:
 		})
 	}
 }
+
+func testGenerate(t *testing.T, in []byte) []byte {
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write(in)
+	}))
+	defer s.Close()
+
+	defs, err := readDefinitions(s.URL)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	symbols, err := parse(defs)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	need := neededTypes(symbols)
```

Comment:
If you wanted, you could also not call neededTypes here, which would generate all the symbols in the input text. That'd let you test generating the struct separately from the command. Your call as to whether you want to stick closer to what the codegen does, or if you want to test specific subsets.